src/reducers/reducer.js

```diff
@@ -0,0 +1,39 @@
+const defaultTasks = [
+  { id: 1, title: 'this is a task' },
+  { id: 2, title: 'this is also a task' },
+  { id: 3, title: 'ignore this one', ignore: true },
+  { id: 5, title: 'this one is archived', snooze: true },
+]
+// const defaultTasks = []
+
+module.exports = function(tasks = defaultTasks, action) {
+  switch (action.type) {
+    case 'TASKS_ADD':
+      return [...tasks, action.task]
+    case 'TASKS_UPDATE':
+      return [action.tasks]
+    case 'TASKS_SELECT':
+      return tasks.map(task => {
+        if (task.id === action.task.id) {
+          return Object.assign({}, task, { isSelected: true })
+        }
+        return task
+      })
+    case 'TASKS_UNSELECT':
+      return tasks.map(task => {
+        if (task.id === action.task.id) {
+          return Object.assign({}, task, { isSelected: false })
```

Comment:
Redux has three main rules http://redux.js.org/docs/introduction/ThreePrinciples.html

Because state can't be mutated, this creates a new object based on the original task and adds the `isSelected` field. It is similar to underscore's `extend` function.